src/utils.js

```diff
@@ -1,7 +1,47 @@
+import * as Errors from 'bitgo/dist/src/errors';
 /**
  * A logger we use to log debugging information to the console
- * @param {String} e 
+ * @param {String} e
  */
 export function logToConsole(e) {
   console.dir(e);
 }
+
+/**
+ * Call the recover() function with recoveryParams, and try multiple key paths for the user key
+ * @param {Object} baseCoin 
+ * @param {Object} recoveryParams 
+ * - backupKey: [encrypted] xprv, or xpub if the xprv is held by a KRS provider
+ * - walletPassphrase: necessary if one of the xprvs is encrypted
+ * - bitgoKey: xpub
+ * - krsProvider: necessary if backup key is held by KRS
+ * - recoveryDestination: target address to send recovered funds to
+ * - scan: the amount of consecutive addresses without unspents to scan through before stopping
+ * - ignoreAddressTypes: (optional) array of AddressTypes to ignore, these are strings defined in Codes.UnspentTypeTcomb
+ *        for example: ['p2shP2wsh', 'p2wsh'] will prevent code from checking for wrapped-segwit and native-segwit chains on the public block explorers 
+ */
+export async function recoverWithKeyPath(baseCoin, recoveryParams) {
+  const userKeyPaths = ['/0/0', '/0'];
+  let recoveryPrebuild;
+
+  for (const path of userKeyPaths) {
+    recoveryParams['userKeyPath'] = path;
+
+    try {
+      recoveryPrebuild = await baseCoin.recover(recoveryParams);
+    } catch (e) {
+      // if this current path we try yields us no inputs to recover, we catch the
+      // error and continue trying the next path
+      if (e.constructor.name !== Errors.ErrorNoInputToRecover.name) {
```

Comment:
`e instanceof ErrorNoInputsToRecover` would be more conventional

Reply:
I was going to ask here if `constructor` could be removed. Otto's suggestion is better I think but out of curiosity is it true that `e.name === Errors.ErrorNoInputToRecover.name` ?

Reply:
`e.name` is "Error", and `Errors.ErrorNoInputToRecover.name` is "ErrorNoInputToRecover". ;) I'm not sure why `e`'s name isn't the child class's though.

Reply:
@matthewzipkin-bitgo  @OttoAllmendinger 
Based on the mozilla's js [guide](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Details_of_the_Object_Model), `instanceOf` has this implementation:
```
function instanceOf(object, constructor) {
   object = object.__proto__;
   while (object != null) {
      if (object == constructor.prototype)
         return true;
      if (typeof object == 'xml') {
        return constructor.prototype == XML.prototype;
      }
      object = object.__proto__;
   }
   return false;
}
```

Since we are comparing the `__proto__` attribute of the error thrown to `prototype` of the error constructor, I printed them out in console:

`e.__proto__` turns out to be `Error` with the name `ErrorNoInputsToRecover`, which means that `e` is only an instance of `Error` and that we cannot use `e instanceOf ErrorNoInputsToRecover` to try to filter this error. I will revert my changes in the last commit.

Reply:
This is the output when I printed out `e.__proto__` and `Errors.ErrorNoInputToRecover.prototype`:
```
BitGoJsError {constructor: ƒ}
constructor: ƒ ErrorNoInputToRecover(message)
arguments: (...)
caller: (...)
length: 1
name: "ErrorNoInputToRecover"
prototype: BitGoJsError
constructor: ƒ ErrorNoInputToRecover(message)
__proto__: Error
__proto__: ƒ BitGoJsError(message)
[[FunctionLocation]]: BitGoJS.min.js:24
[[Scopes]]: Scopes[5]
__proto__: Error
constructor: ƒ BitGoJsError(message)
__proto__: Object
```

Reply:
Ahh I remember this issue now: https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work

Reply:
https://github.com/OttoAllmendinger/BitGoJS/blob/b376b1123740f40d4e8de3cef16c1e92d127be0b/modules/core/src/errors.ts#L153 :thinking: